### PR TITLE
[backup-manager] Rudimentary backup-manager plugin

### PR DESCRIPTION
**Pull request by shtrom**
This is a pretty rough plugin, which just counts the number of recent file (tarballs) in the configured backup directory, but it does the job of alerting when backups haven't run.

Signed-off-by: Olivier Mehani shtrom@ssji.net

### CODE REVIEW

File: plugins/backup/backup-manager [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
```
Comment (sumpfralle) [inline plugins/backup/backup-manager]: Please add a POD-style documentation header here. See the plugin [processes/multimemory](https://github.com/munin-monitoring/contrib/blob/master/plugins/processes/multimemory) for a good example.
Also add the author, a licence and the capabilities.

File: plugins/backup/backup-manager [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# Configuration directives, edit before first use.
+BACKUP_DIR=${backup_dir:-/data/backup}
+ARCHIVE_PATTERN="${archive_pattern:-*.tar.bz2}"
+# How old backups should be considered as non-yound anymore in [days].
+LIFETIME=${lifetime:-2}
+# Critical states will be issued when the number of fresh backups archives is below `backup_number`,
+# and warnings below `backup_number*lifetime`
+CRIT=${backup_number:-1}
+WARN=$((${CRIT}*${LIFETIME}))
```
Comment (sumpfralle) [inline plugins/backup/backup-manager]: You can remove the $ and the curly braces from arithmetic expressions. This should increase readability.

File: plugins/backup/backup-manager [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+graph_category system
+freshcount.label number
+freshcount.critical ${CRIT}:
+freshcount.warning ${WARN}:
+EOF
+        exit 0;;
+esac
+
+printf "freshcount.value "
+find $BACKUP_DIR -name "${ARCHIVE_PATTERN}" -a -mtime -$LIFETIME | wc  -l
```
Comment (sumpfralle) [inline plugins/backup/backup-manager]: Add double quotes at least around BACKUP_DIR (and preferable also around LIFETIME) in order to be able to work with directory names including whitespace.

File: plugins/backup/backup-manager [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+printf "freshcount.extinfo "
+du -sh $BACKUP_DIR
```
Comment (sumpfralle) [inline plugins/backup/backup-manager]: Double quotes, please.

**Comment by sumpfralle** [comment]
That looks good!

I have just one more question: what is the purpose of the lines around `export BM_REPOSITORY_ROOT`? Are they helpful for understanding or using this plugin?

**Comment by shtrom** [comment]
Yes, I'm using this plugin with `backup-manager`, so I thought I might as well give an example configuration that shows how to make both work together.

**Comment by sumpfralle** [comment]
OK - I see.
Please mention (and link) "backup-manager" in front of these example lines (or remove them). The reference to this specific software is not obvious anymore due to the name change of this plugin.

**Pull request merged**